vmdb/app/models/vim_performance_daily.rb

```diff
@@ -59,8 +59,24 @@ def self.find_by_time_profile(*args)
     #
 
     klass = ext_options[:class] || MetricRollup
-    options[:conditions] = klass.merge_conditions(options[:conditions], {:time_profile_id => tp_ids, :capture_interval_name => 'daily'})
-    klass.find(cnt, options)
+    wheres = klass.merge_conditions(options[:conditions], {:time_profile_id => tp_ids, :capture_interval_name => 'daily'})
+
+    # FIXME we should push these up to the callers
+    options.delete :conditions
+    select = options.delete :select
+    order = options.delete :order
+    raise "Unsupported options #{options.keys}" unless options.empty?
+
+    case cnt
+    when :all
+      scope = klass
+      scope = scope.select(select) if select
+      scope = scope.order(order) if order
+      scope = scope.where(wheres) if wheres
+      scope.all
+    else
+      raise "Unsupported finder value #{cnt}"
+    end
```

Comment:
If it's pushed up a a mixin to ActiveRecord::Base a la the ar_* extensions, then this code becomes something like `klass.apply_options(options).to_a`